### DISCUSSION TITLE
Nitro crashes when http call returns an error at application start up.

### DISCUSSION THREAD

**Discussion by avaz** [nuxt/nuxt#16059] [General]
Hi all,

I having this issue which I'm not sure if is a real issue or a misconfiguration. 
My application in the landing page makes an API call which can return 200 or 401. In cases that it returns 401 I wrap the response in a `H3Error` using the `createError` function. This works well in development (`vite`) but if I build the application and run it from `node .output/server/index.mjs` the node process exists if the API returns an error. The thing is that if I avoid this to happen at the start up then later even if the API returns 401/error the `node` process does not crashes and the application continues to run normally and, in my opinion, as expected.

The code showing the issue can found in this repository:
https://github.com/avaz/nuxt3-nitro-startup-error

The basics to have it happening is:

`node 16.13.1`
`nuxt 3.0.0-rc.1` it's happening in the edge as well.

`npx nuxi init nuxt-app`

Files:

`/pages/index.vue`

```vue
<template>
  <h1>{{ user }}</h1>
</template>

<script lang="ts" setup>

import useUser from "~/composables/useUser";

const user = useUser().get()

</script>
```

`/composables/useUser.ts`

```typescript
import {ref} from 'vue';

const state = ref<String>()
export default () => {
  const path = '/api/error';
  function get() {
    try {
      $fetch<String>(path).then(u => state.value = u);
    } catch (e) {
      state.value = null;
    }
    return state;
  }

  return {
    get,
  }
}
```

`/server/api/error.ts`

```typescript
import {defineEventHandler} from 'h3';
import {useApi} from '~/composables/useApi';

export default defineEventHandler(async (event) => {
  return await useApi('status/500');
})
```

`/composables/useApi.ts`

```typescript
import { UseFetchOptions } from "nuxt/app";
import {H3Error, createError} from  'h3';

export const useApi = async (path: string) => {
  const options: UseFetchOptions<any> = {
    baseURL: 'https://httpbin.org',
  }
  const response = await $fetch.raw(path, options).catch(e => e.data);
  let data
  if (response && response.ok) {
    data = response._data;
  }
  else {
    let partial = response;
    if (typeof response === "string") {
      partial = {
        statusMessage: response
      };
    }
    const error: H3Error = createError(partial);
    error.statusCode = 500;
    error.statusMessage = response;
    data = error;

  }
  return data;
}
```

Loading the application at `http://localhost:3000` while in development mode (`vite`) everything works, the error end up appearing in the browser console in this example. However if running the built application from `npm run build` and then `node .output/server/index.mjs` the `node` process exists once the error is returned from the `useApi` composable.

I would say that even if there is something wrong on how to return errors back to `server` calls the `node` process should never exits. 

Someone knows what's wrong?

**Comment by avaz** [comment]
@danielroe sorry to disturb but any idea about this? The node process only crashes at the startup. After startup, even if the http call returns the `H3Error` the server just logs it but don't exits. Any suggestion how to overcome this?

**Comment by LukaZeleznik** [comment]
Has this been resolved?

**Reply by avaz** [reply]
@LukaZeleznik yes, this is not happening anymore in latest nuxt releases.
